lib/ansible/modules/cloud/amazon/ec2_vpc_vpn_facts.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_vpn_facts
+short_description: Gather facts about VPN Connections in AWS.
+description:
+    - Gather facts about VPN Connections in AWS.
+version_added: "2.6"
+requirements: [ boto3 ]
+author: Madhura Naniwadekar(@Madhura-CSI)
+options:
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value.
+        See U(http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpnConnections.html) for possible filters.
+    required: false
+    default: None
+  vpn_connection_ids:
+    description:
+      - Get details of a specific VPN connections using vpn connection ID/IDs. This value should be provided as a list.
+    required: false
+    default: None
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+- name: Gather facts about all vpn connections
+  ec2_vpc_vpn_facts:
+
+- name: Gather facts about a filtered list of vpn connections, based on tags
+  ec2_vpc_vpn_facts:
+    filters:
+      "tag:Name": test-connection
+  register: vpn_conn_facts
+
+- name: Gather facts about vpn connections by specifying connection IDs.
+  ec2_vpc_vpn_facts:
+    filters:
+      vpn-gateway-id: vgw-cbe66beb
+  register: vpn_conn_facts
+'''
+
+RETURN = '''
```

Comment:
Forgive me if I'm wrong but I would assume return values should stay consistently laid out like in other facts modules such as [ec2_asg_facts](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_asg_facts.py). I've had a few occasions recently where I've found lacking documentation so I reckon it should be be laid out the same way.

Reply:
@JohnVonNeumann , thank you for your suggestion. Yes, I do agree with you to keep return values to be consistent with other modules. Through the latest commit, I have corrected these values.
Only for 'customer_gateway_configuration' field, haven't attached sample as it is a big XML string.
Please review and let me know if you would like to see any more changes.